Doc/library/imath.rst

```diff
@@ -0,0 +1,99 @@
+:mod:`imath` --- Mathematical functions for integer numbers
+===========================================================
+
+.. module:: imath
+   :synopsis: Mathematical functions for integer numbers.
+
+.. versionadded:: next
+
+--------------
+
+This module provides access to the mathematical functions for integer arguments.
+These functions accept integers and objects that implement the
+:meth:`~object.__index__` method which is used to convert the object to an integer
+number.  They cannot be used with floating-point numbers or complex
+numbers.
+
+The following functions are provided by this module.  All return values are
+integers.
+
+
+.. function:: comb(n, k)
+
+   Return the number of ways to choose *k* items from *n* items without repetition
+   and without order.
+
+   Evaluates to ``n! / (k! * (n - k)!)`` when ``k <= n`` and evaluates
+   to zero when ``k > n``.
+
+   Also called the binomial coefficient because it is equivalent
+   to the coefficient of k-th term in polynomial expansion of
+   ``(1 + x)ⁿ``.
+
+   Raises :exc:`TypeError` if either of the arguments are not integers.
+   Raises :exc:`ValueError` if either of the arguments are negative.
+
+
+.. function:: factorial(n)
+
+   Return *n* factorial as an integer.  Raises :exc:`ValueError` if *n* is not integral or
+   is negative.
+
+
+.. function:: gcd(a, b)
+
+   Return the greatest common divisor of the specified integer arguments.
+   If any of the arguments is nonzero, then the returned value is the largest
+   positive integer that is a divisor of all arguments.  If all arguments
+   are zero, then the returned value is ``0``.  ``gcd()`` without arguments
+   returns ``0``.
+
+
+.. function:: ilog2(n)
+
+   Return the integer base 2 logarithm of the positive integer *n*. This is the
+   floor of the exact base 2 logarithm root of *n*, or equivalently the
+   greatest integer *k* such that
+   2\ :sup:`k` |nbsp| ≤ |nbsp| *n* |nbsp| < |nbsp| 2\ :sup:`k+1`.
+
+   It is equivalent to ``n.bit_length() - 1`` for positive *n*.
```

Comment:
This function seems redundant.

Also, I think it's better to first just plain moving of functions.  No new features.

Reply:
I'll remove it if you with.

Initially I planned to add more functions (in addition to new functions `isqrt()`, `perm()` and `comb()`). `ilog2()` is similar to `int.bit_length()`, but uses `__index__()`. After our discussion about `math.log()` I think that there is also a use case for 2-argument `ilog()` (to determine the number of decimal digits in a number, for example). But this is a different issue.

Reply:
As any addition to the stdlib, I think it deserves at least own issue.  Lets just settle up the new module and move things properly.